rfcs/text/0013_saved_object_migrations.md

```diff
@@ -212,39 +212,59 @@ Note:
    If none of the aliases exists, this is a new Elasticsearch cluster and no
    migrations are necessary. Create the `.kibana_7.10.0_001` index with the
    following aliases: `.kibana_current` and `.kibana_7.10.0`.
-2. If `.kibana_current` and `.kibana_7.10.0` both exists and are pointing to the same index this version's migration has already been completed.
+2. If the source is a < v6.5 `.kibana` index or < 7.4 `.kibana_task_manager`
```

Comment:
nit: If the user upgraded from < 5.6 to 6.0, they will have a `.kibana` alias pointing at a `.kibana-6` index. However, if they have a fresh deploy < 6.5, then they'll have a bare `.kibana` index. I don't think this has any concrete impact on the algorithm that you're proposing; however, it makes the naming of the index a bit imprecise: `.kibana_pre6.5.0_001`


Source: https://www.elastic.co/guide/en/kibana/6.0/migrating-6.0-index.html contains the manual steps which are equivalent to what the Upgrade Assistant did. 

P.S. I'm primarily relating the information that @tylersmalley gave me on Slack because he's a wealth of knowledge about old upgrades ❤️